tutor/src/models/task-plans/teacher/plan.ts

```diff
@@ -466,11 +466,12 @@ export class TeacherTaskPlan extends BaseModel {
     async grantExtensions(extensions: TaskPlanExtensionData[]) {
         //if new extensions dates are selected for a student who has already an extension, this will update the student previous extended dates
         const grantedExtensions = unionBy(extensions, this.extensions, 'role_id');
-        const updatedExtensions = this.api.request<TaskPlanExtensionData[]>(
+        const updatedTaskPlan = await this.api.request<TeacherTaskPlanData>(
```

Comment:
Maybe the api returned an array of extensions before, but it's returning the full task plan now.